shapelib.vapi

```diff
@@ -0,0 +1,239 @@
+		public unowned string pachFieldType;
+		public unowned string pszHeader;
+		public int nCurrentRecord;
+		public int bCurrentRecordModified;
+		public unowned string pszCurrentRecord;
+		public int bNoHeader;
+		public int bUpdated;
+	}
+
+	[Compact]
+	[CCode (cname = "SHPHandle", free_function = "", cprefix = "SHP")]
+	public class SHPHandle {
+        	[CCode (cname = "SHPOpen")]
+        	public SHPHandle (string pszShapeFile, string pszAccess);
+        	[CCode (cname = "SHPCreate")]
+        	public SHPHandle.create (string pszShapeFile, int nShapeType);
+        	[CCode (cname = "SHPClose")]
+        	public void close ();
+        	[CCode (cname = "SHPCreateTree")]
+        	public unowned ShapeFil.SHPTree create_tree (int nDimension, int nMaxDepth, double padfBoundsMin, double padfBoundsMax);
+        	[CCode (cname = "SHPGetInfo")]
+        	public void get_info (int pnEntities, int pnShapeType, double padfMinBound, double padfMaxBound);
+        	[CCode (cname = "SHPReadObject")]
+        	public unowned ShapeFil.SHPObject read_object (int iShape);
+        	[CCode (cname = "SHPRewindObject")]
+        	public int rewind_object (ShapeFil.SHPObject psObject);
+        	[CCode (cname = "SHPWriteHeader")]
+        	public void write_header (ShapeFil.SHPHandle hSHP);
+        	[CCode (cname = "SHPWriteObject")]
+        	public int write_object (ShapeFil.SHPHandle hSHP, int iShape, ShapeFil.SHPObject psObject);
+	}
+
+	[Compact]
+	public class SHPInfo {
+		public unowned GLib.FileStream fpSHP;
+		public unowned GLib.FileStream fpSHX;
+		public int nShapeType;
+		public int nFileSize;
+		public int nRecords;
+		public int nMaxRecords;
+		public int panRecOffset;
+		public int panRecSize;
+		[CCode (array_length = false)]
+		public unowned double[] adBoundsMin;
+		[CCode (array_length = false)]
+		public unowned double[] adBoundsMax;
+		public int bUpdated;
+		public uint pabyRec;
+		public int nBufSize;
+	}
+
+	[Compact]
+	[CCode (cname = "SHPObject", cprefix = "SHP")]
+  	public class SHPObject {
+        [CCode (cname = "SHPCreateObject")]
+        public SHPObject (int nSHPType, int nShapeId, int nParts, int panPartStart, int panPartType, int nVertices, double padfX, double padfY, double padfZ, double padfM);
+        [CCode (cname = "SHPCreateSimpleObject")]
+        public SHPObject.simple (int nSHPType, int nVertices, double padfX, double padfY, double padfZ);
+		public int nSHPType;
+		public int nShapeId;
+		public int nParts;
+		public int panPartStart;
+		public int panPartType;
+		public int nVertices;
+		public double padfX;
+		public double padfY;
+		public double padfZ;
+		public double padfM;
+		public double dfXMin;
+		public double dfYMin;
+		public double dfZMin;
+		public double dfMMin;
+		public double dfXMax;
+		public double dfYMax;
+		public double dfZMax;
+		public double dfMMax;
+		public int bMeasureIsUsed;
+        public void ComputeExtents ();
+        public void DestroyObject ();
```

Comment:
That looks like a destructor. It could be removed from here and declared with `[CCode (free_function = "SPHDestroyObject")]` instead.